src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_MULTIPLIER = 2;
+    private int threshold;
+    private Node<K,V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K,V>[])new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        putValue(hash(key),key, value);
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
+        int hash = hash(key);
+        int index = getBucketIndex(hash);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if ((node.hash == hash)
+                    && (node.key == key || node.key != null && node.key.equals(key))) {
```

Comment:
sometimes hash could be different. For example key with hashcode 16000 and 16016 will have the same bucket. That's why you should check it by equality 
```suggestion
            if (node.key == key || node.key != null && node.key.equals(key)) {
```